ast.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"bytes"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type subTData struct {
+	subTKey    string     // subtest name key
+	origTDName string     // original test data []struct name
+	newTDName  string     // new test data struct name
+	TD         []ast.Expr // original test data node
+	subTest    ast.Stmt   // Run() function literal node
+}
+
+// return the code of the "test" function from a file
+func getFuncCode(test string, fstr string) string {
+	fset := token.NewFileSet()
+	ppc := parser.ParseComments
+	if file, err := parser.ParseFile(fset, fstr, nil, ppc); err == nil {
+		for _, d := range file.Decls {
+			if f, ok := d.(*ast.FuncDecl); ok && f.Name.Name == test {
```

Comment:
`f` in this case is an [`*ast.File`](https://golang.org/pkg/go/ast/#File) which _I think_ means it will always have a `Name` which, as an `*ast.Ident` should always have a Name string property... Regardless, the fix in the upcoming PR to catch all panics should cover this case.

Reply:
`f` is a `*ast.FuncDecl`: `f, ok := d.(*ast.FuncDecl)`.

Reply:
But a recover would handle it anyway, right.

Reply:
You're totally right! f was a File, but I had to change the variable name to not collide with the function declaration. I was commenting based on an old version of the code still stuck in my head 🙃